storage/kvstore/tests/TESTS/kvstore/general_tests_phase_1/main.cpp

```diff
@@ -101,6 +101,11 @@ static void kvstore_init()
     TEST_ASSERT_EQUAL_ERROR_CODE(0, res);
 
     if (kv_setup == TDBStoreSet) {
+#if COMPONENT_FLASHIAP && !COMPONENT_SPIF && !COMPONENT_QSPIF && !COMPONENT_DATAFLASH && !COMPONENT_SD
```

Comment:
This is cumbersome, but we don't know if the default instance of BlockDevice is FlashIAP or others.